File: electronics/product-service/src/main/java/com/microservice/productservice/Model/Review.java
package com.microservice.productservice.Model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "reviews")
public class Review {
    @Id
    private String commentID;
    private String email;
    private String productID;
    private String title;
    private String content;
    private String dateComment;
    private Integer star;
}

File: ClientPages/src/reducers/review.js
import * as types from './../constants/review';

var initialState = {
    ListReview: [],
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case types.GET_REVIEWS:
            return {
                ...state,
                ListReview: []
            }

        case types.GET_REVIEWS_SUCCESS:
            var { data } = action.payload;

            var { ListReview } = state;

            ListReview = data;
            
            return {
                ...state,
                ListReview: [...ListReview]
            }

        case types.CREATE_REVIEW_SUCCESS:
            var { data } = action.payload;

            var { ListReview } = state;

            ListReview.push({
                commentID : '',
                content : data.content,
                dateComment : data.dateComment,
                email : data.email,
                star : data.star,
                title : data.title
            });

            return {
                ...state,
                ListReview: [...ListReview]
            }
        default: return state;
    }
}

export default reducer;